test/IRGen/class_update_callback_without_fixed_layout.sil

```diff
@@ -320,3 +320,13 @@ bb0(%0 : $ClassWithResilientField):
 // CHECK-NEXT: [[T1:%.*]] = insertvalue %swift.metadata_response [[T0]], [[INT]] [[NEW_STATUS]], 1
 // CHECK-NEXT: ret %swift.metadata_response [[T1]]
 
+// Make sure we cache the construction of fully bound optional types.
+
+// CHECK-LABEL: define internal swiftcc %swift.metadata_response @"$s42class_update_callback_without_fixed_layout21ClassWithResilientRefCMr"
+// CHECK: call swiftcc %swift.metadata_response @"$s16resilient_struct12ResilientRefVSgMa"
+// CHECK: ret
```

Comment:
Hm, is there a reason this still isn't using `instantiateTypeByMangledName` for the fully-instantiated cases?

Reply:
The request is dynamic.

Reply:
We generate the ``type metadata accessor for resilient_struct.ResilientRef?`` it takes the request as argument.

Reply:
The only caller is `` type metadata completion function for class_update_callback_without_fixed_layout.ClassWithResilientRef``:

```
%7 = call swiftcc %swift.metadata_response @"$s16resilient_struct12ResilientRefVSgMa"(i64 319) #5
```

This could call instantiateTypeByMangledName instead of the type metadata accessor maybe ... for more code size savings?

Reply:
Ah, the request being dynamic makes sense. I was mostly just surprised that going through `emitTypeMetadataRef` didn't already use instantiateTypeByMangledName.